src/librustc/session/config.rs

```diff
@@ -720,6 +750,27 @@ pub fn build_configuration(sess: &Session) -> ast::CrateConfig {
     }
     let mut v = user_cfg.into_iter().collect::<Vec<_>>();
     v.extend_from_slice(&default_cfg[..]);
+
+    // Before we touch LLVM, make sure that multithreading is enabled.
+    unsafe {
```

Comment:
Yeah, it'd just be an implicit contract that the driver must initialize LLVM before calling this function.